src/controller/codeValidationController.ts

```diff
@@ -0,0 +1,26 @@
+import type { RequestHandler } from "express";
+import { updateUserAuthStatus } from "../services/databaseServices/updateUserAuthStatus";
+import { findCodeInDb } from "../services/databaseServices/findCodeInDb";
+
+export const codeValidationController: RequestHandler = async (
+  req,
+  res,
+  next
+) => {
+  const { activationCode } = req.params;
+  const foundCode = await findCodeInDb(activationCode);
+  if (foundCode) {
```

Comment:
I think you should "return early" :)

so refactoring the first bit of this code:
```ts
  const foundCode = await findCodeInDb(activationCode);
  if (!foundCode) {
    return res.status(400).json({ message: "Provided auth code is wrong" });
  }
  
  // ... and so on
```